data = "{{{{{{{<!>!>!<u,!>!!!!!!!>!!!>,<<!!g>},<!!!>,<!>,<}>},{{<>},<>}},{{{{}},{{{},{{<!>,<!>,<,'e!!'i'!!<,u>}}},{<ia!!!!!>g!>},<!!u!!!!!>,<a!<!!!>>,{<!!{!!',,u!!!>oo'>}},{{<og!!!o{a>}}}},{}},{{{<!!g!>,<!a!o!!!>!!!{!g!!!>},<o>,{<!i<{}!!!><>}},{}},{},{{{},{<,!>},<!>,<!>},<,!>,<!}!!!>>}},{<{!>,<{oi!a{a!!e!>},<o!!!>>},{<e!','a!!<!i{a},!!!>!>},<>,<{}}!!!>ao!!i}go!!!>,<i!!>}}},{{{<!!!!<{!,e!!!>>},{<ue!>!!!>o'>}},{{{<o'gae!!!>e!>,<g!!o!!!>!>},<>,{<}gu!}o!!u}!>},<!!!>oa<!!,>}},{{{{<'<'!!{!!goeu!!!!ui>},{<,!<!!!>,<!!!>},<!}}!!{!!goug!>,<>,{<<,,!!!>i!!<<!>!!!>,<>}}},{<<a<au!'!!!!!>!>!e!!uge{!!!>},<>}},{{{<!!!!i!!!>!>},<!!!>e,!>!!!!!>!!'g{}>,<g!>},<!!}'!>,<o!!!><>},{{<,!>},<}!!,i!>},<'g<!!!>,<,!!ei'!!a!>},<i>},{{{<{e!!!>,<u!>,<!>},<'!!!>a!!!>}a!!>},{<!!,>}},<!!!>},<,u!!!>,<o>}}},{<!>},<g<!>},<o!>},<!!!>!>>,{{{}},{<!>},<g!>!!}eg!>,<!>},<{g{!!},!!!!!{g<'<>}}}},{{{{<!>,<!!!>o!!}!!!>g!>,<i}}!o!>,<ge!>>}},{{<'i!>e!!!>},<!>!>,<a!!{u!>>,<o!!!>>},{<!>,<>}}},{<o'!>!o!!!>!!!!u!}',!!{!!!>}!!!!!>>}}},{}}}}},{},{{{{{<!!!>{<'!io<,>},{{{}},{{<!!'u<ae>},{{<!>},<!>},<}!>,<}!!{!!!!u,!<!>,<i!<g}<>,{<e!>gi!!'!!a!>},<!!!>o<i!>},<o!!i!>>}},{{}}}},{{{<!>a,g!e>,{<!!!>o!!iu!>},<{!>},<!>io!!{!>,<i<>}}},<u!>,<!!!!<a,}!>'!!'ai>}},{{{<u<!a!>},<!!ga,!!!>,<!>},<eu{}>}},<!!e}!!'!!!>!i!!{!>},<!!!!!>,<<>}},{{{<a!>!>ua!!!>!>},<!!u!>!g!!>,<!>},<{!!!!>},{<!!!>,<!!!!{!>,<!>},<!!g<}!>{,i{'!>,<!>,<>,<{<'<!!!>,<eo}>},{<!!<!!>}},{{},{{<g!>!>},<a,,!>},<}!>!e}<<!a,>}}}},{{{<o'i>,<!aaig!!!>,<!>,<!!!!'!u>},{{{<g<!!!!!!g!!!>u!>,<!!!>>}},<ia!!a>}},{{},{{<!,u!>},<}!!u!!{!>,<>},{{<!>},<}{o}!>},<!>},<ui!>,<!!<!>!>},<>,{<'!!'o}}i!!!!!>!!{<,e>}}}},{{{{<o!!!>!!!>},<>}}},{<oo!>i!>},<!!!>!>},<}i'<!i,!>},<!!'>},{<a,o!!!>a!!!>>,{<aae!u<!>},<e!>}o!!!>}!>a'ia>,<g!,i,e!'!!e!>,<!g!!!>i!>},<>}}}}},{{<!>uu<g>,{<'}!!!>!!!><{!!{g,!>''o>}},{<u!>},<{!!!>{!!!>,<o'o{g>,<o!{g{!{!>},<!>,<ei>},{<!!!>,<!{}a!!e<>,{<!a!!i<ua!!o!!!>,<!!g!>}!>},<i'}a!>!>>,{<!>,<<!>ag{gg!>e!>,<<<>}}}}},{{{{{<a,!!!!!>!>e}!!,i}a,!!u}e,a{>}}}},{<!>,<>,{<,gug!!!>a!!e!!!>>}},{{<ie!!eg!><!>,<'>}}},{{},{{{{{},<!!e>}},{{<{!!g{}e!>}u!!!>u!g!>},<ue>}},{}},{{<aau!>,<!>!>,<}g!!!!a!o!!!!!!}<!!{!>,<'!><g>,{<!!{!{!>,<e!!',!!!!eue!>>,{<a!>,!!i}!!'}!!}!!ueu!!{}!!!!!>{u>}}},{{{{<!!,oo!>,!!}a!!g!!!>!>>,{<!>}<g!>},<!!!><!!!>,<ue<!>'>}},{<{,!>!'a<<!>,<a!>!!!>!<}<<>}},{<{!!!!!>},<,!>'<!>e}!>,<i!!!>'!!>}},{<!o!!!!!>o!>},<u<!>,<!>{}!!a!!a>}}},{{{<eo!>,<!!g}ee!>},<ua!>},<!!!>iu>}},{{<}'ag!!!a!'{!!!>!!!>!>!>o!a!!}!!!!!!!>,>},{}}},{{{{{<!!!!!>!,<}!!!!!>,<a!<<a!!!>>,<!!u!>e!o!!!!!>!>},<!>e!!!!!>io!ie'!!>},{<!>,<g!>},<<ou!>},<ei!>,<i!!!>,<<{o!!!!!>{>},{<!>,<ao>}},{{{{{<<o!>,<ga!>o!,o!!!>!gai'!!'>}}},{{<!!gea'}aei!!a!!!!<iag,>},{<>}},{{{<!>i!}e,io<u!!!!!>,<!>},<i!!}i>},{<!ag!!a}!!}>}},{{<>}},{{{<!u!>!},eg!!!>ue,!>!>{!>},<{'}i!>>},{<'a,g!>a>,<!!!>!>,<!!iag!i!!!!!>,<{>},{<ie}}'ga!>,!>,<>,{<>}}}}}},{{<}!{i!!}!!!>i,!>,<}ga!!i!o>,{{<e}>},{<ie!'!!<!>},<i!!!>,<!!u!!{>}}}},{{{},{<o!>,<!>,<!>,<!!!>!!!!!!}!>{!!i!>oe'>}}},{{{<!!e!!o{!>,<ei!>,<,u<o!!!!!><>,<!!e!!!!!>>}},{{{{}}},{{<a!!!>},<o<!uo!>'!{!>},<,},!>,<e<>}},{{<!aig>},{<a!>a'{<>}}}}}},{{{}},{<!!>,{<!>},<!!!>},<{!}e>}}},{{<,!>},<!!ai'i!g!>,<!>},<g{gu!>>,<<!>,<!>},<!!'!a!>},<!!!>!>},<'!!!a}'>},{{},{},{<,!!e!>},<!>},<!!!>!>},<!a!>},<g!!o!>u,<,,!!>}},{{<<a!>a}!a!!!>},!!e'{!!!!'!!!>!!!!o>},{{{<,>},<}!!!>g>},<io!!g<i<{}!'!>,<,{!>>}}},{{{<!>!>!>!!g>,<!>!!!>!!<'}'>},{{<o!!{a!!!>'g!!>,{<!!!>o!!!>,<o!!!!i{<'}!!!<e,>}},{}},{{{<a!>o<o!>u!}i!>},<!>,<,g!!a,''!>},<>}}}},{{{{{{{<<!>},<>},{<o!!!!!>},<!>''o}o!!<gu!>},<u!!!>e!!!!!>>}},{<!>,<a,ae}',{}ua<u!>},<,eu>}}}}},{{{<!!!>,<}'<<{{!!e!!!>{i}!>},<!>,>,<}!!<i!!!>},<{!>},!!!!i!<!'!!,!!{u<!>>},{{<g!>,<'g<!!!!!>g{!>>,<ea}g!o<<'ui!!<o>},{{{{<}o>}}}},{{}}},{{},{}}},{{<a!g!>},<}!!!!!!!>u!>,<<a!!igo,o!,!!!>>},<!!!>!!!!ai!!!><!!}!>g!>g'ge>},{{{<e,!!'!!!,!>,<}au!>,<!><gi',<g>,<!>,<{g'}i!>},<}!>,<,>},<>},{{<'e>},{}},{{{}},{<e!!i>}}}},{{<,>,{<ou!!!>!,}'ig!!e>}},{<o!u!!!!u!!!!<!!}!>},<<!>,<!>,<!!{!!ue!>eie>,{}},{{{<!o!>'o!!u!>g}'!>!!!!!>!>},<}g>,<}g,e!!!!!>g!>},<!>>}},{<a!>,<i!!!>,<!>u!}!!u!!>},{<},!!,ioi!!!!!>!>'!!!>u>}}}},{{<!!!oa>},{{{<{a!>},<!!!>!!i{{ug>,{{}}},{<i<{,,!!}!!u!>}u>,<!!iu}!!<ua!!!!!>>}},{{},<!!>},{<!>},<!!!>!>},<i}<!!!>!>},<{,>}}}}},{{},{<!>!>!{o!!!>!>},<!>},<<!>,<{!!'!!!><!!u>,<g'>}},{{{<{!>!>!>!>!>!>},<!>e!!!!u!!!!!>u'>},{},{{{<>}},{<!!!>{<!>,<{e!!{!!!>gu!>aa!!!>,<>,{<!>g>}},{{},<oa!>>}}},{},{{{},{<{e'!>,<ioo!!!!!>,!!!>>}},{{{<,{g!!!>},<eg'e,iu>},<u!!!>!!!,!!iuuuegaug{>},<}>}}},{{{{<'!>},<e!>!g!!!>!!!!a!!{a!{!ieii!!!>!o>},{{<!>,<!>o{!!!>},<!!!><!>},<i!!!!>}}},{{<a!!'o!!>},{{<!!!!a!!ia!!<}eoug!'!>},<'>}}}},{{{<!!!!,}!!!>},<!!!!!>,<>}},{{<{!!!!!>g!>},<!>,<i!>},<!>,<!>a!>},<!>,<{,!!!>!!!!uo>},<,!!u!}>},{<o!!!!uo'e!}a<o!!!,<!!og{>,{}}}}}}}}},{{{{},{{<<<>},<!!!>{!!!!i{!!!>oo>}},{{},{{{<ea'ue!>},<!>{!}>,<,>}}}}},{{<>,{}},{{{<!!!>'}{o!>,<!!!>!i!!!>,<!>,<!!'g!!!>!!!>}>},{{}}},<!!!>g!!!!<!!'!u<!>,<>}},{{{},{<!!,!!{!!!>},<!!!>>}},{<!!!!!!!>o!!>,{{<u}i!>,i!!,i}e>},{<!>},<!!g!}!>,<!g!>},<!!!>!!<a!!a!>},<!!!>>}}}},{}},{{{{{<e{!!!><}ei},eg!!!>!>},<'>,{<<!!!!!>,e!e}!!!>},<i'ug!>,<>}},{{}},{<g!i!>!!{!!{o!>},<!i{!>,<>,{<ei,>}}},{{{<!!!><!{a'oo>}},{<a!>a!'e!>,<!>,<>}},{{{},<!>},<!!!i!!!>},<!>,<,o!>},<}!!!>{!!e!!!!u!!<<!!}!>},<!>,<}>},{<ae!!!>{>}}},{{},{<!>!>,<!'!,ge!>o,!!!>!>,<e,}!!<o,>}},{<g{!>,<<i!!!!o{<o>,{<!!}>}}},{{{{<!!!>}!>,<ig!!igeoieaaii!!!>u>},{<!!}e!!e>,<!!!!<aie!!o!>},<!>,<!>,<!!!!!>!!<!eu,!>u>}},{},{{<!'!>,<<!auo>},{<!>,<>}}},{{<!>},<'g!a!!!!{!>!>,<o!>},<u!>!{}!>>},<o'e{'ag!!!>>}},{{<},g!o!>},<,,!!!>,<!>!!!>!!!>a!>>,{}},{{<<egu>},{{{<!!,>,<>},{{<!!!><!}a!i!!}!>},<o!!g''e>},{}},{{}}},{<g!>},<!>,<!'!g{!!!!!>},<!>},<>,{<!!!>e!!!!!>au!>},<{ue>}},{{{{<o,!>!!!>!!ua!>!!!>o!}}>},{}},<i!>},<<e!u{eua!e'},>}}}}},{{<ee!!!>'i!>auu!!{{!>g>,{}},{{<u'!<!>},<!>},<!!!>'!!>},<a!!!>ge!}!>}>}}}},{{{{<!!!>!g<,!!,ua{!}'a'!a!>,<!>,<>},{{<a'!{,gg{u!!{<!>u!>{i!>,<!!!>,a>},{{},<!>!!!>!!!>},<!'!!,i,ii!!!!!!!><'!!!!oi,!!>}},{{{}}}},{{<,!!!>,<,u!!!'ia,>,<!!i!>},<!!{<>},{{{{<!!!>i{e!>,<i,!>},<u!>,<!a,o!>},<>}},{<!>,<e'>,<ee!!!>a!!!>},<<!>>}}},{<o{<e!!!!!>!!e'!>,<gu>,{<!!!>{g!!!!!>!!!>gu{g>}}},{{{<!!geo!!a}<{'>,{<u>}},{{<e,,>,{{{<o!!!>o!g!>{{{!i'!><!>,<!>,<>},<a'!!!!!,u!!e<!a,>},<g!!ue>}},{},{}}},{},{{{<!>a!!!>!!!>!!oo'o!!<}{ig!>,<'u!>},<e>},<ig!!!>ou'!>,<!!!>},<g<<{i>},{{{{<!!!>o<<g{!!}a'g,,!!!>g!!!><>}},{{{<!>,<eau>}},<eee},!!eo!<}>}},<u!!}oe>}}},{{}}},{{{{{<ai!>,<'{>}},{<e!!!>},<!>,<!><!!!>a>}}},{{<!!<!!!>},<!>},<a<<!>!g!>,<e{>,{<<g!>,<>}},{{<,o!>g!<u!>,<!>,<u}!!!!>},{{<o'>},{<,!!!>!o>}}},{{},{{<,!>!>},<>,{<!!!!!!go!!g!!a!>,<!>},<>}},{{{<!!}!!!!!>o!!!>{!!>}},{{<!!!!!>,<e'u!>},<'u!>},<u!>!!e,eg>,<o!a>},<euoo!!!>>},{{{{<!!!>'o{>},{{},<e!><>},{{},{<<e,!!!>!>},<}!>,<!>},<'{ee!>,<e,>}}},{<{{<g{u<!!!!i!!}!>,<''{g!!!>>},{{<!!i!>,<g!!{'<e!e{!!!>,<<{!>},<!>!{!!!>!>},<>},{<,iu!!'},ao!!au>}}},{<!!!>i,>},{<ee!>i<!!u{!!!>oa'}>,{<,e'>}}}},{{<!>},<<,,>,{<}>}},{<u}!>,o{!<<,!>,<e,!!a>}}},{{<!g!>,<e!>,<o!>!>},<a}!!eu!>},<!!<>,{}}}}},{{{<>}},{},{<g!ui!>},<!{<!<!uig'!>u,ga!!!>'>,{<!!i!>,<geae}a>}}},{{{{}},{{{<!>},<gei!o!!!>,<{!>,<!!!>e,a>},<!!<>},{<!>ggii'e{e,!!!>,!>,<{>,<!>!!!>,<o!!}{e!!!>i!>,<!>!!!>!<a<{!>},<!>},<>},{{<!>},<!>'eg!!!!!>,g}',g!!!>!!e!!uu>}}},{}},{{{{}},{}},{{}},{<!>},<}!>},<!>},<!>,<o!<!>gu{{ea!>,<!}!>>,{{<<<!>},<o,!!!>{>}}}}}},{{{<<!>},<!>,<>},{{{{<!>},<g!!u!>}!>,<,!>,<!>,<,,>},{<!!!>{!>},<g!>,<!!!!!i,!>},<!!!>},<},i!><>}},{}},{{<!>!!!>g}!!g{!>},<!!ug!!{e}au!!>},<i<{!!!>e'!!!>eu{aggg{a!!!>!!!>>},{{{}},{}}}},{{{<!!'!>o!>},<i!gi',{a>},<>},{},{{<'i!!{i}<ii'{oi'a}u!!ei!!!u>},{<>}}},{{<'!!!>!!a<u!>,<}!i},g>,{{{<!>!>,<!!!>a!>!>},<!e!!!>!><g>},<!>,<e!!'u<<<!e!'>}}}}},{{{{{{<{',g!>},<>,<}!uaa!>,<<eia<ie>}}},{{{<<g,i'>},{<u'!>e<!!aa!!!>,<!,!'>,{<!!!>,<uu,!}!!,uia'e!!ie!!i!a!!!!{!>},<>}},{{<!u!!!>},<!>,<!>},<!>},<g>}}},{{{}}},{{<,<{e!!!o!>},<!!!>'i!>,<!!!>a}>},{<a!!!>},<i'!,!!{!>},<!!}e>,{}}}}},{{{<{!>!!!>,>}},{{<!>eo<'u!a,'iu<!i!}!>},<>},{<}<!>},<!>},<>}}},{{{<!!!!!>!!!!<'ga!<!>,<g!>},<o!!o>,<}e<ou!>oa!}!!!>!!}iu}!>>},{}},{{<>}}},{{<e'i>},{{<<!>!!,>}},{{<a>}}}},{{{{{{{<>},<g!!{!!!!!!!>u!!!>a!!,a'!}}!!'!!!>>},{<!>,<!>},<ug>}},{{},<!>!>},<!!a!'ee!>,<}eoiao!!!!!!!eu>}},{{<g!!,!!!>},<!!}!!!>i!!!>,<>}}},{<i{<!!a!!!>!>},<,!!!!'>,<!><!!,!uo!!}<{uoo,'!!!!!>{>}},{{<e>,{}},{{<g<!>},<!>u!>!>,<i!!a!!oeo}}og!!g>}},{<!!!,!!a!>},<i!>!!!>},<!u<!!!>>,<{g!!au'gu<ua!>,<,,>}},{{{{}},{<<ei{!>,<>}},{{{<,!g!!g!><}ai!!!!!>!>!!!>!,ui>},{}},{},{}}},{{}}},{{{{<!!}o}!!g}<!!}'!!aao}!!!!e>,<>},{{{<!!!>i!>,<!!!>!>u'!>,<,>,<!!g!>,<a!!,!!e!!!>age!!!>!>>},{{<!e},eouua!!oooeogoo!!!>>,{<eu,!>!>,<!!!>!>}{<!>,<!!!>!>!!g!!!,gi'a>}},{<o'!>},<!!!>e!!!!}''oo!>,<g<>,<}o!>'!!g!>},<!>},<g!}e!!!>!!},>}},{<}i,<!!<u>,<{!!!!<g}>}},{<g,,i,e{!!!>,<'i<!!!>},<>,<<!!!>}u!!go!>!g>}}},{{{},{{<a!>},<<}'o!>,<ug,!!!><!>,<!'<ae!>>},<a!>,<uoo!'!!i!g!!!!!>!><'gu!!!!!>,<a}>},{<!!!>,!>},<,!!o<i!>,<iu!!,i!!!,{>}},{{},{{<a!>},<'!>},<!ea!>!!,g>},{<!>},<>}}},{{{<a!!!>!!<!>},<a!!ou!>},<e,!!i!uo!!!>!!{'>},<!>},<o{}g}!>,<>},{{{{},{{{},{<!>},<!>},<}!>!!o<>}},{{{<!g'<g!!!!!>a!>,<e>}},<!!!>}!!g!!u!!!>},<}!!!>},<,g}<!!i!!!>>}}},<!!!>!>euo!!,u!!!><!>!!<eo!>,<a!>,<>},{{},{{{<!e!!<!>},<!!!>,!ou{!,>}}}}},{<!!!>!e!>ee!!g!!!!!>aia'{!>,<>,<<e!!!>{!>!>},<<}>}}}},{{{{{{<!!'!<>}},{<{!g<a!>}a{<e!!g!!},}'>}}},{{{{<!>!!'!>,<,u!!!!i!!a!>!!e>}},{<,{!>,<!!!>!a!>,<{!!!>!!>}},{{{<,ao!!<{!>{'!>},<u!!!>!!!>o!!!,!>,<g<>}}}},{{{{{}},<!!!!g!>},<!>},<!>,<o!!<!>},<!!o{!>,<<oau!!!!}>}},{<!!!>!>!>,<<!>e,>,{}},{<,!!!>!>,<}!!eao<{,!!ag!>,<!!!>!>},<ogg!!g>}}}}}},{{{{<o!!!>!>},<'!!>},{<}{'!>'!!g{!>u!'go{ga!!>}},{{}},{{<!!!}!>},<{{!>!!u!i>},{{{}},{}}}},{}}}},{{{{{{<>},{<>}},{{{<!!!>,!!!>!!u>},{<!ua!!!>!!>,{}}}},{{<e!>},<!>},<!!{ia!!!!!>>}}},{{}},{<}!a'}!>,<<!o!!eu!>},<!!uu'!!!>g!>},<!!e>}},{},{{},{{<!o<!g'g!!u<!!iu!>!>},<<ai>},<>}},{{{<!{ii!!}i!>,<a>,<!>!>,<!!,>}},{<!>'o,!!i!!<oo!>,<u'!>i!!!>'>},{{{},<>},<i!!!!,}!!!!aue>}}},{{{{{<!!a!!!>ga!!!><geuoe!g>},{<au,o}!>,<>}},{{{<!>!>,<a,a!!!!!>'!e!ug!>,<!!>}},{<!!!a}!>,<o!<{!{!>,<!!!>}a,g!!u!>},<i!!}o>}}},{<g>},{{{<a{e<>}}}},{{{{{{<o!>},<!>!!a!>},<{ao!,i<!!!'{>}}}}}},{{{<!!!!!>!>!{e}{!>},<!!!!!>,<,}oao!><>}}},{{{},{<u!!!!'u!>!!e!!ou'u'u}>},{{{<,ag{o!!!eug>},{<a!>'!!!>,<{!!!e!!!!ge}!>},<<o{{!!!>},<u{u!>>}},{<>,<}'i,!!!>,<!>},<>},{{},{{<g!>},<}!>,<o<!>,,!>,<ueu'!>,<>}}}}},{<!>,<!'uig<,u>,<!>,<<{g!>},<!g!>,<>}}}},{{{{{}},{{<e{{!ee!>,<a!>,<{!>!!,!!i'u!!!>,<!!!>!!!>>},{{<!!a>},<i<ee!'e{'!!'uoi'!>}>}},{{{{<a!!!!!>>}}},{{{<!>},<!!!>!>,<!!!>!!<!!!>!a!>,<<a!>,<!>!!!>!!!>{g,>},{}}}}}},{{{{<!!!!e!>,<<!!!>!>},<!}o>},<!!,'g,!!!>,!>,<!>},<!!o,!!!>},<!i<!>,<!>},<{>},{{<e!>},<!!g{!>},<>},<a!!>},{{<!>,<e',ae!!!>g!!<{,o!ai!!a}{!>>,{{<g!!!>,!>},<<!{!>},<u!>,<!>,<>},<!iu'!>},<!!i!!i!!!>!!!>!>!>,<i!!u,!><a>}},{{<o!,ee<>}}}},{{{{<}!>,<<a!>oaa''!>!!!!!>!!o>},{<'<,!<go,!>},<!>},<ai!>!!<'o,!>>}}},{{<uea!>,<{'!>},<o!g,e!!u}{o!>},<{}!o>},{{<!>}<i!!o!>,<!uoo>},{<,!>!!!>'g!!!>!><,ouo>}},{{<ea,guoao!>,<g<>}}},{{{<!!ai,o''!!<e<{g<eo!!!>>,{<g!!!e}!!!>!!!u!!!>!!<!!!!!>},<{u}a>}},{<',!>,<o!>,<!>,<iu,!>,<>}}}},{{{{{<!>,<<!!!>!!{!!!!,>},<!!!ea!!!>g>}}},{{{{{<!>},<!>ue'u'!!g,!!!>}>}},{<!!ua!>,<!><!!,o!>,<}}{>}}},{{},{<ui!>},<!!}u!!!>,'!!uu{!!{<e<!!e!>,<!>>}},{{<i!!!>ua!!!>!!!>!!!>u!!'u!!!>},<!e>},{<!,!}!!e<!!'!>!!a!!!!!>!>},<,>}},{{{{<!!!>!!{}u!!!!ii!!!>ga'>,{<ie,!}!>i'!ee'!>,<a!e!!!>{>}},{{<g!!!>o!!!{o!>,<ge!>,<g!'{iaogi>,<!!,e>},{},{}},{{<u<'<}>},{{},<!>o'!!oao!>!!!>},<!{!>},<>}}},{{<!>},<!>i,!!o>},{{}}}},{{},{{<!!o!!iu'>},<!!!>{a<eao>}},{}}}}},{{{{<}{>},<a!!!>!>,<u'!>,<!>,<{iio,'!!uig>}},{{{{<!>,<i!!o>},{<{{g,'!!,!>,<!!}u!!a}g!!!>g<!>},<>}},{{{{<g!!!!g!e'!>,<!>'<!a!e<>}}},{<!,!>},<{gi'i>}},{{{<<!>,<a!a{ao>},{<,>}},{},{{{<iui!!!>,<<},!}io!!,o{>},{{<,g!!!!,e!>,<!>},<,!!!>e!>i!>,<u!!{i!!,!!!>},<>},<!}<}!!e!>,<!'{i!!!>},<!i!!!>!'{,!!i>}},{<u!!}!a!!!>!>,<!!!>}!!au!>},<a>,{}}}}}},{{{<>},{{<!iu<i!>{u,!o!>},<g!>},<>}}},{<<!>!>,<!!!!!>!a>,<g!!!>e!>,!>},<g!u!!!>>}},{}}},{{{{{{<u!!,ag<u!!'i,{>,<!!{>},{}},{}},{{{<uio!>,<'!{io{!>>}},{<!>},<!!o!>,<!',,}'!}i<e!o<oa!>,<!!!>e>,{{<<g!>i!!!>e'!!!>,<>}}},{{<'!i>},{{},<{o!,>},{{{<!>,<ei!>!!,<!>,<!>!>,<{a>},<!<>},{{<!>},<ag!!}o,!!!>u!!!>,o>,<uu!>!>},<>},<!eai!!!>!!!>,<i!>,<!}og!}!!!>o',!>},<!>},<>}}}},{}},{{{{{{<}!!!'a!!!!og'}!!ee!!,!>,<!gg>}},{{<'<!uug!!!u'!!uua,!gi!!,!>,<!!!>,<{>}},{<!u!!!>!o!!u>}},{{{<u,!!!!!>}!>},<!!!>!!g}!!!>o,'!>,<!>!}o!i>},<!>!>},<eeei'!>!!u!>,<!!g,,!o!>'o!!,'>},{}}},{{<!>,<{e}!!!>e>},{{<!!<e!!!>!!g>,<a'g!!<!>},<a!!i!>eg}!!<<!!i!>},<!!!>},<>},{<!!!>'!>},<auu!!!!!!!>!>},<'>}}}},{{{<!>,<egg!>,<ig>}},{<!!!>a!{!,<!>},<'<!>o'!>g}!eea!>,<!!!!g>,{<!!!>!!<o{!>}!>},<}}!>,<,!e{!<e!>,<!>!aa>}}},{{{{{<o,!>},<i!>>},<!>,<,'>},{{<!><!o!a!!!>},<!!!!!>,g!,g,}!{a!>,<!>,<!>>}},{{{},{{{<g'g'!!i<!ea!!!>,<i,!!!>e{u!}>},<!!!><u{!>!!!!!>i{g!!ggu>},{}},{{<e!>},<a!!!>!>!!!>>}}},{{},<!a,!>,<,ug!ao!eog!o<>}}}},{{{{<!>},<>},{{{{{{<e!>},<go},!>},<!{a>}},<i!!'!o!a!!!>},<a!>,<!!!!>},<!!i},<i!!!!<'!>,<'>}},{}}},{<!i,!i!>'}!!!!!>g{}'g!!!!>,{<ouia!!!!!>,<'!!!>',{!>!!}!!u!!<'!>,<!>},<>}},{{{{}},{<!>,<a}!!!>}>}}}},{{},{{<e}a!!}aagg!!aa!!{!u!!!!!>>},{{},{{{{<!><!>!!!>u,!!i{!>},<'!>ao}{>,<!!!!a,o!!o'>},{{<'!!!>,!!eiu>},<}i!!!>,!!o}!!!!!!!>,<ao}u>}},{{{{{{<!>o<'!!e!!!{}!>},<!a>,<u!'!!u!>},<!>g}!e!!!>a}{!>,<>},<!!!>!!!>>}},{}},{{<!!!>e!>},<>}},{{{<!,!>!!o}<'}}>}},{{<!!!!!!!!'!!!!,'!!!>!>,<},a!!>},{<>}},{{<{ug>}}}},{{},<!a!!!>g}!!!>},<!>},<>},{{{<<>}}}},{{{<'a!!!>eouu!>,<!!!>,<uo!!a!!i>}},{}}},{<o!!!>a!>oe<!>},<<!!>,<!!!a,aui!>gge!>'i}!>},<!a>}},{<i!>},<o!!!>,<a!>},<>,<}au!>},<!!!>u!>,<u!!!>},<e!!}a!!''!{>}}}}},{},{{{{<<!!!!!>{{!!!>!!oe!>,<g!<o!>,<a>},{{<!ge!>,<>}}}},{{<!>!>,<{!!!>u!>'!>},<>,{{<o!>},<,e>}}},{<<eaoee'!!!>,<a!!!>!!e!!!>,<!!!!!>g!>o!>},<e>,{}},{<!>,<'!!!<!>'>,{{<!!a!>},<!!!>!>ug>}}}},{{<ag!>,<'>,<!!!>,<uo!>,<ou,!!!>!>},<!!!>,<'u!!!>!>,<!!!>,<u>},{{{<ua!>,<}u!>,<!!!>!!!>!!!>!!!!'>},{<u'e}!!!>!!!,!!o!u!>},<g<!!g!>,<>}},{{<!>},<<o{u'>},{}}},{<!!g!>u<ai!!!>!>,<!>}!!'o!>,<>,{}}}}}},{},{{{{{<iia!!eig!!!>gg>},{<o,!>e!>!!{!!!>g}i!>,>,{<<{!!g'!>>}}}},{{<<!!},!!>},{{{},{{{},{<!,a!!!>},<e!<!!!!!>u!>u>}},{}}}}},{{},{<e!!!>{,<a!>},<!{,e!o!!!>!!>}}}}},{{{{<'<!!!>},<!!!>!!!!{!!!>u>}},{<a>,{}}},{},{{{<o!>},<}!>,<!!!>{!!!>,<!!a>},{<u!!'i!<!>!!!!!>,<!>!!!>!>,<!!>}},{{<e!>,<u!>e>},{<!!{!!!agu<!!o!!>,{<o'!!!>!!!>!!!>},<ui!>,<>}}}},{{{<!!!}!!!>{o'!>!>gog,o'<o!>a>}}}},{{{{<!g!!!>{i,oe'!!,au!><!!!>},<u,>},{{{<ei!!!>!'i'a<>},<!!,!}!!!><a!!}'au!>g!>,<!!!ge''>}}},{{{<{o}!>,<gg!>,<<,'{u,<}<!}>},<{u!!!>!!!>!!e<>},{<>}},{{},{{},{<u{!!!!>,{<!}u!>},<'e!!!>!>,<!>!!!!,!!!!!>!!!>!>},<!>,<!!}!,>}}},{{},<e!!<geui!!eo>}},{{{<!'{!!!>'i{a!g'!!!!!!!>>},<{u!>,<}<ig!!!>i>},{{<!!}!>,<{!!a!>i!!,!>,<<!!aa!!>,<>},{<!!{e!!<}>}},{{<!!!>'!,>}}}},{{<o!!!>!>,<!!!>a!!!>!>,<'!>!>},<}!e'o!>,<a>,<!!!>,<!,g!,!>i}!{!>,<!>},<<<a<>},{{<!>,<<!>'<!!!><!!go>,{<<,a}{g!!!>'!>},<<<!!!>o!!a!>,<i>}}},{{{},<giaeg}!>u{!!!>!!g{uoaa!>},<!!!>!>,<>},{{<{e!<}!<!!!>!!!>,!!'!!o{,!!!>,<{!a!!g>},{<!>,<}!>},<!u!>},<o!!<{u>}},{<o,!>,<i!}!!!!!>,<!!aa!>},<u<!!!>},<>,<{o!a!!!!e!>!,!>!i!>!i>}}},{{{},<!!aou},!!!!!!i!>u!>,<}u>},{<!!,!<!g!!ea!!!!!>o}!i}i{!!!>},<!>,<'!<'>,{<!u{u!>!>,<,!>},<!>{<g>,{{}}}},{{},{{<i!!!>>}}}}}},{{{{{<!>},<u,!!!ui!!!><!uu!>},<!!!>e!!!{o!!a>}},{{}},{<o'<!!<>}},{{},{{{{<!g!iig,!!!>!!ea!!u',!!!>>},{}},<!!!>},<ua!!u!!!uio!!!>a>}}}},{{{{{{<!!!!'!!!!io,e,!!i!!!>'u>},<!!e!>},i!>},<e!>!!!>,<',}ee!>>},{<a!g}!!oaoe!!!>iug{>,{}},{{{{{}}},{{<<!!{<!>euao>}}},{{{<g}>,{<!>eg!!!>{e>}},{<g}g!>},<'!>,<'!!!>i!o!>i}!!!>!a>}},{{<!>}!>},<!!!>{!>>}},{{}}},{{<!>,<g!,!>},<!>ua<!!!>},<au!'g!!,u{!>},<!!!>>},{<!>e>}}}}},{{{{}},{<!!!>!g!>},<!,!>,<!>!!!>!!!>u!!!>,g!,u!!!>!e}!!!>e>}},{{{{<!{!!}!>},<,'i<!>,<!{ee!!g>}},<!ag!!{!!!>,<}i'{!>},<',>},{{<}!>{!>},<a!,<!>},<u>}}},{<o}u!!<'<!>,<i}!!i!<!i>}}},{{{{<>}},{}}},{{{<e!>!!<ee!!!>}ga!!i!}!!g!!uai>}},{{},{{{}},{<'og<e!!!!{}'!!!>,<go<g!>>}},{}},{{<!!!>>,{{}}},{{<,!!!>uuu!>},<}{,u<}'{>}},{}},{{<i!!!!}u<!!!>'!>'!!!>,!!!>!!!>,<}o!>,<,i>,{<}!!!>'!>,<!!!>!!{!!u}'u}guo>,{<!>u}g!>,<euo,!>,<ea,>,{{},<!!uia!!!!,<a!>,<u<!!!!!>e!!>}}}},{{<,ie>}},{<e!!!>g!>,<!>ue!!!>,!!!!!>},<i!}!>!!!!!>i!>>,{{{<<agu,'!>ua!!!!{i!!{g>,{}}}}}}},{{{{<!>!!!!>},{{<!!a<{<!>},<>},<,i>},{<!!!>'!u'g!!{!!e}>,{{}}}},{{<!>,<!!e!!!}!>,<u}i'o,!!u>,<!!i!>!>,<!!!!!!!!!>>},{},{{<!!!!,!>,<ue!!!,u!e!!{!!!>!>,<}'e>}}},{{{<<>}},{{<o{oi!>},<g!!!!g!e!!!>!>,<{,}u'!!!>>},{<ia!!!>!>},<}a'!>}>}},{{},{{<'!>},<!!i!!!>!!!>},<!!!>u},'!>,<o<i{ou!!>,{}},{{<!>,<!!a{{>,{<{!,o{<!>},<!!ea!>,<,!>},<!>!>,<'g!>,<oi!}>}}}}}}},{},{{{<!>,<!>,<<!!!<!!!!!>,<u,ae>},{<}i!!au,'!>},<!>},<!!!!!!!>!>>,<g!>},<!<g!>,<!>,uio!>,<!>},<!!!>!!!><>},{}},{{{{{{<!>},<}g>},{<a!>,<!!!>>}},<{g!!!>!!!>!!!>},<}!!!>},<!!!>g'!!!!!>!>,<}>},{<!u!!!!!>!,<!!!>!>!!,<e{>}},{<!{!!}{o,u}a{!aa!>,!!!!}!!!>},<a>,<!>u!!!><a!>!!'},{oua!>},{!!!!!>>}},{{{<uua}<!!!>{,!!oe<>},<!>ue,!!!>!!!>},<!!u!!{e!!!>>},{{<gga!!!!',,',!>},<u{a>},{<ei!!}!!!!!!!>{!!!>a!auo!>,<>}},{<!!e!>!!!>!>},<!>},<!>,<uoo!!!>,<iiui!!{!>,<!!!!!>,<!>},<>,{}}},{<g!!!>!!!>!>,<!>},<!!!>!>}g>,<'!{!!!>!!!>g!!!!!!!!o>}},{{{}}}},{{},{{{<io}g{{eg'!{<!!!>ui!>e>}}}}},{{{{{<{}<u!!!!!!!>o!!!>'!!}!!!uo!'!!!!!>>},<>},{<!!!>>,{}},{{{{{<>}},{<!g!!!>g!!oo!>}a,!!,uu!gg!!!>!>},<>}},{{<!!!>o>}},{{{{<>,<!>},<}!u!!!>,<{o}}g!>},<e!!!!gi!!!>,<}a>}},{{<i}!!!>{!!,iee!!!!!!{!>,<!>>}},{{},{<}!!!>!!!!!>{!>,<>,<!>},<e<!!,!!!!e!>,<!!!>>}}},{{},<!!!>},<<{ue}!oagg!>oi<u!!i!>,<!i>},{{},{<!!u!'}i!!{!!!>,<!>,<!!!>},<!o'!!!>>,{<}!!!>!!!>,<agu!,>,{{},{<!o{!!!><{o>}}}},{{<>,{<!io!!'}e{!!!!!>!!!>!!!>!'}!!!>!aui>}}}}}}}},{{{{{},{<!gi!><ua!>!>},<!!e>}}},{{{<!>,<!>{{!>,<{}gig}!>},<!>},<<<{au>}},{<!<}!!a>},{{{{<!>,<!!!>,ai',!!!>igi!>}g>}}}}}},{{<!!!>}!!!<u'o','!>!!!>!>},<!!!!!>},<!>},<!!{>},{{{<'!i<}!!e>},{<!!!!!>oo!!<!u'!ogie!!!>!!!>ee{>}},{<>}}},{{{<o!!gg!!!!!!}{<e>},{<!!{!!!>>}},{<!!!>!>i!u!><!!!>e!!a!!!>i!!!>eu'>,{<i!!!>,<!!}!!!iu<a>}},{{},{<''u!>u!>},<,!!ig!!!>{'}'!>},<!!!>i>}}},{{<!>,<!!{!!!>!>,<!<}!>,<>,<!>,<!!!>!!{a!!ao!i!<<u!}!!!!!>,<>},{},{<!!!!!!!!ui!i!!ogaai}!'ue!!i!>,<gi!!!!!!!<>}}}},{{<!>},<}'!i'a}>,<!!!!!>,<{!ueo{},!!!>,<!!!>'!>},<o!>,<o,}>},{<'!!<>,{<!>},<<!!!>!!!>ee!!<{!>,<o{{{>}},{{{<!>!!,>}},{<!!e}!>},<{u>},{{{{<!>},<,!>i{g!o!>!>},<!>},<!><>},{{<!>},<eu!!g!>},<<!>!!ig>}}}}}}},{{{<!o!'aa}{!!{<ag>}}}}},{{{{<',eua!>{>,<g!!!!!!<g!>,<!!'}u!>,<>}},{<!>,<''!!<{!>,<}}o!>},<ae>}},{{<oi!!!>!!!>!>},<!!!!!>o>,{}},{{{{}}},{{{<!!o>,<!o>},{}},{{{<}u!>,<>},{<!!}a<!!!>uo{,!>,<!a!!{{}!!!>,<!!!!!>!>,<!>>}},{<a{!a{g!!!!,o!>},<iu{{!>a!{'o>,{{<!!!!'!!!>,<!a!!{,o!!!!!><'u>},{<!i}!!og!!}<!!ee!!!>!!o!>},<!!g!>,<,>}}},{{<e!!}i}!i>},<o!>},<a'u!!!>g!!eg!!!>!>,<!>,<!iu{!'!>>}},{{<i!>,<!>,<eg>},<e<uo>}},{{{<!!!>e'!>},<''<egue>},{{<}!!!o>}}},{{}}}}},{{<}!!!>},<u!!'e!!,!!!>,<!!!>!u!>,<!!g!>},<}a!>,<!>>},{{<!!!>!!o!>},<!!o!>,<}!!!>i>},{{<!>,<!>,<>}}}},{{{{<o{g}>},{<u!>,<,!!u>,<!>u!!!o<!io>}},{{}},{{{<g{g}'!>,<!}!!<!!!>,!!ga!!!>}i>}},{{<!>{!!!>!!!>},<!>,<!>!!!>',e>},<ag}a{e!>}{<ae<!>},<ge>}}},{{<!>!>},<!<a!!o>},{<!!e{!!ga<{i'!o<<!>},<!>,<!!!>o!!i!!!e>}},{{},{{{},{<g!>,<!!!>a>}},{}}}}},{{{{{<'!!!>ie!!!,a!>}'!>,<!!<e!a!>!>>,{<,,}!>},<ua!!'!!!>i>}},{{},{<!>,<<!!!>},<,!!e>}},{}},{{<,ou!o!!ge!!!>i{!>,<<>,{}},{<!!!!!>go!>},<{!!!>gig!!!>>,{<{!>},<g!>},<!>a!!!>>}}}},{{{<!!}'ao<{u!i<!>,,!!<g!!!!ei>},{<!!!>a}i!>,<o!>},<!!!!!>!>},<>,<g!>!!!gug!>!>!>},<!!!!!>i!!<!>,<u!!{>},{}},{{<!>!>,<!!e>}},{{{{{<!>},<!>,<!<a!!!!!>o}!!{,!>,<!>,<u!>,<u>}},<,!>},<!!}e!}}!!!!!!!>!>!a'!!ge!!}>}},{{}}}},{{{<gi!>},<ie!i>},<,o!>},<!!eie!!!>!>,<<o<g,!>!>g>},{<{!>,<!!!>!!!>g',!>},<>,<!>!!!>o!!!oe!!,!>g!>},<,>}}}}}}"

# current score of group
g_score = 0

# total score of data
g_total = 0

# count of characters in garbage
g_count = 0

# ignore characters until garbage ends
garbage = False

# skip a charcter after an ignore
skip = False

for d in data:
    
    # actually ignore the next character
    if(skip==True):
        #print d, "\tskip"
        skip = False
        continue
    
    #ignore character inside garbage
    elif(d=='!' and garbage==True):
        #print d, "ignore"
        # just skip next character
        skip = True
    
    # start of garbage
    elif(d=='<' and garbage==False):
        #print d, "start garbage"
        garbage = True
    
    # end of garbage
    elif(d=='>' and garbage==True):
        #print d, "end garbage"
        garbage = False
    
    # start of group (only outside of garbage)
    elif (d=='{' and garbage==False):
        #print d, "start group"
        # increase score
        g_score = g_score + 1
    
    # end of group (only outside of garbage)
    elif (d=='}' and garbage==False):
        #print d, "end group"
        # add score to total
        g_total = g_total + g_score
        
        # decrease score
        g_score = g_score - 1
    
    # everything else could be a count
    else:
        # but really it only counts if it's inside garbage
        if (garbage):
            g_count = g_count + 1
            #print d+'\t\t\t\t\tg '+str(garbage)+' '+str(g_count)
        
print "garbage count is:", g_count
# correct answer is 6346